### ISSUE TITLE
Add the possibility to turn off displaying element status and type on braille displays

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#7887]
### Steps to reproduce:
1. Connect a braille display to NVDA (in my case, Hims Braillesense U2)
2. Open a program where objects have status and type messages (i.e. browser or mail client)
3. Navigate through the objects and try to get the relevant information as productive as possible

### Expected behavior:
In braille settings, it should be possible to turn off element types and status to be displayed on a braille display. In this case, NVDA should report the status of an object only by speech feedback (read or unread, has attachment, high priority etc.). The same for element type (i.e. Heading level 3, Link etc.).
On the braille display, only the label should be displayed (i.e. the name of the sender, the subject or the label of a link or a heading.
An exception should be included for checked or unchecked objects (i.e. radio buttons etc.). This kind of status should be displayed on the braille display even if "display object status and type" is turned off in braille settings.

### Actual behavior:
In such programs, NVDA displays the status of objects (has attachment etc.) and the type of elements (link, heading etc.) on the braille display. This is ambiguous when you work with speech and braille simoultanuously. Often you want to check some things quite fast and read them again on the braille display. But the fact that NVDA displays any information which has already been provided by speech takes lots of braille cells and makes the use of braille displays in this case quite unproductive.

### System configuration:
NVDA version:
2017.4

NVDA Installed or portable:
Both

Other information:

Windows version:
Windows 10 Build 16.299.64

Name and version of other software in use when reproducing the issue:
Browsers, Mail clients, Windows installer, other programs

### Other questions:

Does the issue still occur after restarting your PC?
Yes

Have you tried any other versions of NVDA?
Yes, 2017.1, 2017.3

**Comment by feerrenrut** [comment]
@Adriani90 Please could you update the description with a specific example. Please link to a specific web page and mention your browser, specify the input, and note the output you get. In this case braille and speech are likely required. Then to clarify, give the output you would like. 

Giving a specific example saves us from having to guess / interpret your description, which reduces the time it takes for people to understand the issue, and also reduces the error rate when trying to understand the issue.

**Comment by Adriani90** [comment]
@feerrenrut, this issue is highly related to braille output, not to input. It is related to NVDA's behavior when displaying information on a braille display. It is not related ot a single website or mail client, but to all websites and other programs where objects have status and where the speech feedback would report them. For example on every website where there is a link, on a braille display the first three braille cells are being used for lnk which stands for link, eve though you have got the information that it is a link by speech feedback. If there is a mail which has an attachment, then almost half of the braille cells are being used to display "has attachment" even though you have got that information by speech feedback. In conclusion, I want to read the sender and subject of that e-mail or the label of the link on the braille display, but not the status if it has an attachment or the type of an element.

**Comment by feerrenrut** [comment]
As has been discussed on other issues, I can appreciate that this applies to many applications, and that it is about output. Giving an explicit detailed example ensures that we are all on the same page, and are talking about the same thing. When I mention input, I am referring the actions you take with your computer in order to create the specific example. When I mention output, I refer to the braille / speech that you get. Braille can be translated back into text, speech also can be translated back into text.

An specific detailed example is a concise way of highlighting the problem / feature.

**Comment by Adriani90** [comment]
I have tried to edit the issue accordingly. In my view, it is very clear because it occurs on every braille display on every website and in every other similar aplication. It is not easy to meet developer's expectations when it comes to formulate an issue. Some developers understand it quite well as it is now, but I can imagine that some developers do not. Especially new developers. So, I hope we can find a common way to make such issues as you imagine them to be. But be aware that sometimes discussions on clarifying an issue are very valuable especially when the user has not as much knowledge as a developer. And in fact, the time invested in such discussions is not time lost. But I agree with you that such discussions must be avoided as often as possible.

**Comment by feerrenrut** [comment]
Yes, it's about communication. Its common for a problem to be very apparent to the person raising the issue, and for others to have similar issues, and all agree that there is a problem. But it's often not until you bring in the specifics that people realise they are all talking about different things. This is often worse when there are language barriers and technical terms involved, as is often the case with NVDA development. 

The point of this is not about avoiding discussion, its to ensure we are all starting from the same place with our discussion. Github issues get very long and confusing when people are talking about different things. The reason I mention time, is that when there is not a clear description, it is often someone from NV Access that will go and reproduce the problem, and provide the specifics. This can be quite time consuming, and takes time away from developing NVDA.

**Comment by feerrenrut** [comment]
I'm not set up to use Outlook so here is my interpretation of the description based on web browser usage, not sure there are settings outside of document formatting that you are interested in, hence the importance of a specific example.

## Steps to reproduce
1. Ensure you have a Braille display connected
1. Ensure you have "Headings" checked in the "Elements" group of the "Document Formatting" settings dialog
1. Open the following link with Chrome (version `63.0.3239.84`) https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_header
1. Press down arrow

## Actual
Speech:
```
heading  level 3
Less important heading here
```
Braille:
```
H3 less important heading here
```

## Expected
Speech:
```
heading  level 3
Less important heading here
```
Braille, notice "H3" is missing:
```
less important heading here
```

## Discussion
I know that I can disable reporting of headings in the document formatting dialog, but affects both braille and speech. It would be nice to separate document formatting options for speech and braille output. This allows those who use both braille and speech to make better usage of braille cells.

**Comment by Adriani90** [comment]
This is a great description of one part of the issue. The thing is, to separate braille and speech regarding document formating settings would not be enough. We would save some braille cells in many cases, but in many cases status information will be displayed though (maybe not in browsers but in other applications).
In outlook for example, the status „has attachment“ is fully displayed on the braille display. So in some cases we are talking about many braille cells which could be used for more relevant information.

Von meinem iPhone gesendet

**Comment by feerrenrut** [comment]
Ok, as far as I am aware there is no way to turn off information such as "has attachment" in outlook at the moment. The difficulty here is that it requires a human decision that information like "has attachment" is supplementary or "extra" information. This means that for each control in each application a developer will need to determine how important that control is.

I can understand why this is desired. It is possible to make rules about the verbosity of metadata, but "has attachment" is not metadata about some text. It is visible information on the screen, it is treated as text itself.

**Comment by Adriani90** [comment]
Hmm, this is quite strange. Because when I change NVDA's language to german, the word "has attachment" is being reported in german "hat Anlage" even though Outlook is set in english. So, the text "has attachment" and "hat Anlage" is not displayed on the screen. I understand your point that in the past there must have been a developer who traduced the sign for "has attachment" to be reported as text by NVDA and that this is quite heavy to turn it off for braille since we would have to do it individually for each application which has information like this.

I would create a pull request for some applications I know of if I knew how to separate such reports between speech and braille. I will try some things for outlook to see if it works in my local copy of NVDA.

**Comment by Adriani90** [comment]
But for meta data like document formating and type of elements it would be possible to create a radio button to turn it off for braille displays. Right?

**Comment by feerrenrut** [comment]
Yes, the "has attachment" text comes from the appModule in `source/appModules/outlook.py`.

> But for meta data like document formating and type of elements it would be possible to create a radio button to turn it off for braille displays. Right?

Yes, I think the place to start on this would be to consider how to implement it for the options in the document formatting dialog. Then we would need to work out how to extend this to appModules.
